### PR TITLE
#92948 check permissions on 'go to case' in tasklist

### PR DESCRIPTION
**Pull request by Klaas-Ritense**
Screenshot 2023-12-22 at 10 45 58

### CODE REVIEW

File: projects/valtimo/components/src/lib/components/carbon-list/carbon-list.component.ts [hunk 1]
```diff
@@ -366,34 +369,50 @@ export class CarbonListComponent<T> implements OnInit, AfterViewInit, OnDestroy
...
+                data: '',
+                key: '',
+                viewType: ViewType.ACTION,
+                sortable: false,
+              }),
+            ];
+          }
+
+          const showLocks = !!this._items?.find(item => !!item.locked)?.locked;
+          if (showLocks) {
```
Comment (teodor-ritense) [inline projects/valtimo/components/src/lib/components/carbon-list/carbon-list.component.ts]:
```suggestion
          if (this._items?.any(item => item.locked)) {
```

File: projects/valtimo/components/src/lib/components/carbon-list/carbon-list.component.ts [hunk 2]
```diff
@@ -154,6 +156,7 @@ export class CarbonListComponent<T> implements OnInit, AfterViewInit, OnDestroy
   @Input() showSelectionColumn = false;
   @Input() striped = false;
   @Input() hideToolbar = false;
+  @Input() translationKey = '';
```
Comment (teodor-ritense) [inline projects/valtimo/components/src/lib/components/carbon-list/carbon-list.component.ts]: Change this to something more specific, like 'lockedTooltip'

File: projects/valtimo/components/src/lib/components/carbon-list/carbon-list.component.html [hunk 3]
```diff
@@ -107,3 +107,14 @@
     </cds-overflow-menu-option>
   </cds-overflow-menu>
 </ng-template>
+
+<ng-template #rowDisabled let-data="data">
+  <div *ngIf="data.locked" class="disabled">
+    <span
+      class="float-right badge badge-pill badge-secondary bg-grey"
+      ngbTooltip="{{ translationKey + '.rowLocked' | translate }}"
```
Comment (teodor-ritense) [inline projects/valtimo/components/src/lib/components/carbon-list/carbon-list.component.html]: Leave this just as a 'lockTooltip' variable instead of doing translation in the table. If it's done like this, then we limit implementation teams in the way that they structure their translations

File: projects/valtimo/dossier/src/lib/components/dossier-list/dossier-list.component.html [hunk 4]
```diff
@@ -74,6 +74,7 @@
     [items]="obs.documentItems"
     [pagination]="pagination"
     paginationIdentifier="dossierList"
+    translationKey="dossier"
```
Comment (teodor-ritense) [inline projects/valtimo/dossier/src/lib/components/dossier-list/dossier-list.component.html]: Update this accordingly, after doing the changes before

File: projects/valtimo/dossier/src/lib/components/dossier-list/dossier-list.component.ts [hunk 5]
```diff
@@ -343,6 +348,27 @@ export class DossierListComponent implements OnInit, OnDestroy {
...
+          )
+        ).pipe(defaultIfEmpty([] as boolean[])),
+      ])
+    ),
+    map(([res, documentsAuthorization]) => {
+      res.documents.content.forEach((document, index) => {
+        document.locked = !documentsAuthorization[index];
+      });
+      return res;
+    }),
```
Comment (teodor-ritense) [inline projects/valtimo/dossier/src/lib/components/dossier-list/dossier-list.component.ts]:
```suggestion
    map(([res, documentsAuthorization]) => 
      res.documents.content.map((document, index) => ({
        ...document,
        locked: !documentsAuthorization[index];
      }))
    ),
```

File: projects/valtimo/task/src/lib/task-list/task-list.component.html [hunk 6]
```diff
@@ -49,22 +49,27 @@
...
+      (click)="$event.stopPropagation()"
+    >
+      {{ 'task-list.goToCase' | translate }}
+    </a>
+  </div>
+  <div *ngIf="!tasks[currentTaskType].tasks[data.index].caseLocked">
+    <a
+      class="float-left cds--link"
+      cdsLink
+      href="javascript:void(0)"
```
Comment (teodor-ritense) [inline projects/valtimo/task/src/lib/task-list/task-list.component.html:68]: Why is this here?

Comment (Klaas-Ritense) [reply projects/valtimo/task/src/lib/task-list/task-list.component.html:68]: To make sure the color of the link is correct

File: projects/valtimo/dossier/src/lib/components/dossier-list/dossier-list.component.html (hunk 4 above)
Comment (Klaas-Ritense) [reply projects/valtimo/dossier/src/lib/components/dossier-list/dossier-list.component.html]: Changed

File: projects/valtimo/components/src/lib/components/carbon-list/carbon-list.component.ts (hunk 2 above)
Comment (Klaas-Ritense) [reply projects/valtimo/components/src/lib/components/carbon-list/carbon-list.component.ts]: Changed

File: projects/valtimo/components/src/lib/components/carbon-list/carbon-list.component.html (hunk 3 above)
Comment (Klaas-Ritense) [reply projects/valtimo/components/src/lib/components/carbon-list/carbon-list.component.html]: Changed

**Review by teodor-ritense — APPROVED** [review]
